src/actions/user.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * @file user.js
+ * Exports actions related to user.
+ */
+
+import { call, put, takeLatest } from 'redux-saga/effects';
+
+import {
+  USER_LOG_IN,
+  USER_LOG_OUT,
+  USER_SET_ROLE,
+  USER_ROLE_EDITOR
+} from '../constants';
+import { getAccessToken, getCsrfToken } from '../lib/api';
+import actionGenerator from '../lib/actionGenerator';
+
+/**
+ * Dispatches actions that grab an access and CSRF token for the specified user.
+ *
+ * @param {object} payload - Payload for this saga action.
+ * @param {string} payload.username - Name of the user that should be authenticated.
+ * @param {string} payload.password - Password of user that should be authenticated.
+ */
+export function* userLogIn({ username, password }) {
+  yield* actionGenerator(USER_LOG_IN, function* userLogInHandler() {
```

Comment:
This is the most complicated saga in the project so far. It relies on the `actionGenerator` method, which I'll also comment on.